[FindMyMouse]Add additional settings

jaimecbernardo opened the pull request:
## Summary of the Pull Request

**What is this about:**
Add more settings to Find My Mouse.

**What is include in the PR:** 
Settings for:
- Background color
- Spotlight color
- Overlay opacity
- Spotlight radius
- How bigger the spotlight is when the animation starts
- Animation duration (0 is instant)

![image](https://user-images.githubusercontent.com/26118718/143061709-1d03ce93-d501-4a1e-a18e-3302c9c6ad01.png)

**How does someone test / validate:** 
Change some settings and see them applied to Find My Mouse.

## Quality Checklist

- [x] **Linked issue:** #13924 #14539
- [ ] **Communication:** I've discussed this with core contributors in the issue. 
- [ ] **Tests:** Added/updated and all pass
- [ ] **Installer:** Added/updated and all pass
- [x] **Localization:** All end user facing strings can be localized
- [ ] **Docs:** Added/ updated
- [x] **Binaries:** Any new files are added to WXS / YML
   - [x] No new binaries
   - [ ] [YML for signing](https://github.com/microsoft/PowerToys/blob/main/.pipelines/pipeline.user.windows.yml#L68) for new binaries
   - [ ] [WXS for installer](https://github.com/microsoft/PowerToys/blob/main/installer/PowerToysSetup/Product.wxs) for new binaries

htcfreek commented:
@jaimecbernardo 
Great. Love this options. What about the "don't activate on fullscreen windows setting"?

htcfreek commented:
@jaimecbernardo 
Some small nits:
- Can we add the unit to the description of `Overlay opacity` and `imitial zoom`? Would help users to understand the setting.
- We use an other naming layout. Only the first letter should be big. (Example: `Spotlight Color` must be `Spotlight color`)

jaimecbernardo commented:
> @jaimecbernardo Great. Love this options. What about the "don't activate on fullscreen windows setting"?

Focusing on appearance for this PR.
But thanks for reminding :)

jaimecbernardo commented:
> * Can we add the unit to the description of `Overlay opacity` and `imitial zoom`? Would help users to understand the setting.


@htcfreek  What do you mean with unit?

htcfreek commented:
> > * Can we add the unit to the description of `Overlay opacity` and `imitial zoom`? Would help users to understand the setting.
> 
> 
> @htcfreek  What do you mean with unit?

What do I set here? Opacity in percent from 0 to 100? Zoom in pixel from x toy or in percentage from 0 to 100?

Jay-o-Way commented:
- Text like "How bigger the spotlight is" doesn't sound very correct.
- Opacity: `0-255` or `0-100%`? (Percentage feels more natural to me)
- no `max` for delay?
Should we mention the bounds for values in the description or not?

niels9001 commented:
@jaimecbernardo Just a general thought: I wonder if we should auto collapse the appearance and behavior settings for Mouse utilities by default? The extensive customization options are great, but might 'overwhelm' users when opening the page. Especially since we have multiple utilities on a single page.

jaimecbernardo commented:
@htcfreek

> What do I set here? Opacity in percent from 0 to 100? Zoom in pixel from x to y or in percentage from 0 to 100?

You're right, it's an alpha value. Perhaps I'll make it 100.
Regarding the zoom, it's how many times, meaning 1x, 2x, 3x, 4x (default is 9x). I made it configurable mostly for photo sensitive users for who that initial flash when the spotlight is big causes some pain. The idea here is that it doesn't need to be too configurable.

> Maybe a number box is better for the zoom setting?

Doesn't need to be too configurable, although perhaps I'll raise the maximum a bit for this issue: https://github.com/microsoft/PowerToys/issues/14539

stefansjfw commented:
Background color colorpicker doesn't work for me. Whatever I select it stays black

jaimecbernardo commented:
> Background color colorpicker doesn't work for me. Whatever I select it stays black

@stefansjfw ,
You have to up brightness (black is described as 0 brightness in this control, which affects all other colors):
![image](https://user-images.githubusercontent.com/26118718/142998678-1e02cdc5-449e-4c3a-bf6a-48a8b6774cc2.png)

stefansjfw commented:
> > Background color colorpicker doesn't work for me. Whatever I select it stays black
> 
> @stefansjfw , You have to up brightness (black is described as 0 brightness in this control, which affects all other colors): ![image](https://user-images.githubusercontent.com/26118718/142998678-1e02cdc5-449e-4c3a-bf6a-48a8b6774cc2.png)

Ok. It works :)

jaimecbernardo commented:
@Jay-o-Way 

> * Text like "How bigger the spotlight is" doesn't sound very correct.

Changed to "how many times larger".
Regarding the zoom, it's how many times, meaning 1x, 2x, 3x, 4x (default is 9x). I made it configurable mostly for photo sensitive users for who that initial flash when the spotlight is big causes some pain. The idea here is that it doesn't need to be too configurable.

> * we could group 'background' and 'opacity' together, then all the spotlight settings are together too.

Placed opacity at the beginning.

> * Opacity: `0-255` or `0-100%`? (Percentage feels more natural; zero percent makes it kind of useless; to me)

Changed to percent. Made the minimum 1%.

> * no `max` for delay?

What would the maximum delay here be? I agree it is somewhat non-sensical to have very large numbers, but I also saw no reason to limit this.

> Should we mention the bounds for values in the description or not?

The reason, in my opinion, for not adding bounds to the description is that then those would have to go through the localization cycle if we changed those, which means we could have some outdated

jaimecbernardo commented:
> @jaimecbernardo Just a general thought: I wonder if we should auto collapse the appearance and behavior settings for Mouse utilities by default? The extensive customization options are great, but might 'overwhelm' users when opening the page. Especially since we have multiple utilities on a single page.

Sounds like a good change. Applied it.

jaimecbernardo commented:
Thanks for the feedback. I've committed change or answered to the suggestions so far. Updating description screenshot as well. PTAL

htcfreek commented on src/settings-ui/Microsoft.PowerToys.Settings.UI/Strings/en-us/Resources.resw:
```diff
@@ -1733,6 +1733,33 @@ From there, simply click on a Markdown file, PDF file or SVG icon in the File Ex
+    <value>Spotlight initial zoom</value>
+  </data>
+  <data name="MouseUtils_FindMyMouse_SpotlightInitialZoom.Description" xml:space="preserve">
+    <value>How many times larger the spotlight is when the animation starts</value>
```
What about "Spotlight zoom factor at animation start"?

jaimecbernardo replied:
Sounds good.

htcfreek commented:
As I understood the zoom factor setting is important for photosensitive users. Should we add a warning bar under the setting, that they should not use a too small value?

stefansjfw reviewed (approved):
Implementation looks good. Works good as well

jaimecbernardo commented:
> As I understood the zoom factor setting is important for photosensitive users. Should we add a warning bar under the setting, that they should not use a too small value?

In this case the issue is with big values.  A warning makes sense. Added the change and updating the image in the description as well.

stefansjfw commented:
> > As I understood the zoom factor setting is important for photosensitive users. Should we add a warning bar under the setting, that they should not use a too small value?
> 
> In this case the issue is with big values. A warning makes sense. Added the change and updating the image in the description as well.

Nice!

htcfreek commented:
> > > As I understood the zoom factor setting is important for photosensitive users. Should we add a warning bar under the setting, that they should not use a too small value?
> > 
> > In this case the issue is with big values. A warning makes sense. Added the change and updating the image in the description as well.
> 
> Nice!

Does it make sense to bind the visibility on the slider value. For example when it's higher than 50% show the warning? 🤔

niels9001 commented:
Why not add the warning in the Description instead of an InfoBar (for the sake of leaving out as much as UI as possible, since the list is growing :))?

**Spotlight initial zoom**
A smaller size is recommended for photo sensitive users

niels9001 commented on src/settings-ui/Microsoft.PowerToys.Settings.UI/Strings/en-us/Resources.resw:
```diff
@@ -1733,6 +1733,37 @@ From there, simply click on a Markdown file, PDF file or SVG icon in the File Ex
+    <value>Overlay opacity</value>
+  </data>
+  <data name="MouseUtils_FindMyMouse_OverlayOpacity.Description" xml:space="preserve">
+    <value>Opacity percentage for the overlay</value>
```
Isn't the title already self-explainatory? Doesn't add a lot and basically says the same thing twice.

jaimecbernardo replied:
Both @Jay-o-Way and @htcfreek asked for it:
- https://github.com/microsoft/PowerToys/pull/14590#issuecomment-976192150
- https://github.com/microsoft/PowerToys/pull/14590#issuecomment-976245073

jaimecbernardo replied:
@niels9001  , so perhaps if I just add make the description "Percentage"? Does this make sense?

Jay-o-Way replied:
What about simply adding the unit to the end of the first text?
`Spotlight radius (px/pixels)`
`Spotlight inital zoom (factor)`
`Overlay opacity (%/percentage)`
`Animation duration (ms/milliseconds)`

\#KISS (keep it short and simple)

niels9001 replied:
I think this should be resolved in a way that it's flexible and can be used across all modules (e.g. setting the opacity for Shortcut Guide overlay, or Zone Opacity for Fancy Zones).

One elegant solution could be adding it to the tooltip:
![image](https://user-images.githubusercontent.com/9866362/143048971-e50a8deb-8e60-4ebe-b6ea-e4e27028c2fb.png)

Other option is to add it after the input control. E.g.:
[ NumberBox ] ms
[ Slider ] %

I'll create a tracking issue for this so we can keep this as is and @jaimecbernado can continue doing awesome things :)!

jaimecbernardo replied:
Thank you 😄 
So I've removed the description for now.
PTAL and approve if you think it's ready 😉 
![image](https://user-images.githubusercontent.com/26118718/143051725-80550eca-48f1-4edc-83e3-10d5fc6b7131.png)

jaimecbernardo commented:
> Why not add the warning in the Description instead of an InfoBar (for the sake of leaving out as much as UI as possible, since the list is growing :))? This will also not flag us for Accesibility related things as the InfoBar should be announced when selecting the previous setting.
> 
> **Spotlight initial zoom** A smaller size is recommended for photo sensitive users

@niels9001 , like this?

![image](https://user-images.githubusercontent.com/26118718/143032198-18f622f8-8dc6-404b-a2d4-d7a99ae426cf.png)

htcfreek commented on src/settings-ui/Microsoft.PowerToys.Settings.UI/Strings/en-us/Resources.resw:
```diff
@@ -1733,6 +1733,33 @@ From there, simply click on a Markdown file, PDF file or SVG icon in the File Ex
+    <value>Spotlight initial zoom</value>
+  </data>
+  <data name="MouseUtils_FindMyMouse_SpotlightInitialZoom.Description" xml:space="preserve">
+    <value>Spotlight zoom factor at animation start. A smaller size is recommended for photo sensitive users</value>
```
Either `factor` or `size`.

jaimecbernardo replied:
Thank you! Committed the change.

crutkas commented:
I don’t want to make that claim without having the low vision board for photo sensitive. Let’s please remove that statement and have them test it. There are tons of nuances here so I don’t want to make assumptions

jaimecbernardo commented:
> I don’t want to make that claim without having the low vision board for photo sensitive. Let’s please remove that statement and have them test it. There are tons of nuances here so I don’t want to make assumptions

@crutkas ,
I've committed the change and updated the screenshot in the PR's description.

jaimecbernardo commented:
@niels9001 , Waiting for your approval and I think this can go in after that.
Trying to get this out for 0.51 RC.
Thank you, in advance.

niels9001 reviewed (approved):
LGTM!
